ibis/backends/flink/registry.py

```diff
@@ -281,10 +251,37 @@ def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
     return f"{table_column_translated} [ {index_translated} + 1 ]"
 
 
-def _day_of_week_index(
-    translator: ExprTranslator, op: ops.temporal.DayOfWeekIndex
-) -> str:
-    arg = translator.translate(op.arg)
+def _array_length(translator: ExprTranslator, op: ops.arrays.ArrayLength) -> str:
+    return f"CARDINALITY({translator.translate(op.arg)})"
+
+
+def _json_get_item(translator: ExprTranslator, op: ops.json.JSONGetItem) -> str:
+    arg_translated = translator.translate(op.arg)
+    if op.index.dtype.is_integer():
+        query_path = f"$[{op.index.value}]"
+    else:  # is string
+        query_path = f"$.{op.index.value}"
+
+    return (
+        f"JSON_QUERY({arg_translated}, '{query_path}' WITH CONDITIONAL ARRAY WRAPPER)"
+    )
+
+
+def _map(translator: ExprTranslator, op: ops.maps.Map) -> str:
+    key_array = translator.translate(op.keys)
+    value_array = translator.translate(op.values)
+
+    return f"MAP_FROM_ARRAYS({key_array}, {value_array})"
+
+
+def _map_get(translator: ExprTranslator, op: ops.maps.MapGet) -> str:
+    map_ = translator.translate(op.arg)
+    key = translator.translate(op.key)
+    return f"{map_} '[' {key} ']'"
+
+
+def _day_of_week_index(translator: ExprTranslator, op: ops.Node) -> str:
+    arg = translator.translate(op.args[0])
```

Comment:
Remember to use the name of the argument and not `op.args[index]`.

Reply:
Thanks for the reminder, fixed it.